src/mgr/ActivePyModules.cc

```diff
@@ -297,6 +297,48 @@ PyObject *ActivePyModules::get_python(const std::string &what)
       pg_map.dump_osd_stats(&f);
     });
     return f.get();
+  } else if (what == "osd_pool_stats") {
+    int64_t poolid = -ENOENT;
+    string pool_name;
+    PyFormatter f;
+    cluster_state.with_pgmap([&](const PGMap& pg_map) {
+      return cluster_state.with_osdmap([&](const OSDMap& osdmap) {
+	f.open_array_section("pool_stats");
+        for (auto &p : osdmap.get_pools()) {
+	  poolid = p.first;
+	  pool_name = osdmap.get_pool_name(poolid);
+
+	  f.open_object_section("pool");
```

Comment:
Can you refactor this to share the code between this path and the "osd pool stats" command handling?  This is pretty much the same as all the `if(f)` paths in the other place.

I'm imagining something like a `dump_pool_stats` function on PGMap that takes the poolid and pool_name as an argument.